developer_workflow.md

```diff
@@ -1,41 +1,87 @@
+
+## Card Review Elements
+
+Understand what the client is asking for.  The client should explain not only the business purpose of the change but any set expectations.
+
+What are the business requirements? Any existing specifications, technical requirements, mock-ups, or other supporting documents we should review?
+
+If the card represents a bug or defect, ask how it can be reproduced as well as the severity of the defect.
+
+Are there target dates either for initial review or going live with the change?  If so, find out more about the nature of those dates.  Are they fixed or flexible?  What's the consequence of missing the date?  Are they vital enough that the scope of the requested work should shift?
+
+Ask questions to clarify anything that you either don't understand or isn't clear in what has been covered so far.
+
+Explore edge case scenarios that may not have been covered to see how the client wants those resolved.
+
+Clarify where in the system these changes should take place.
+
+Identify any related items that may match similar business functionality but may not have been mentioned explicitly by the client.
+
+Bring up any technical concerns with the requested change.  Making sure the client understands the costs and added complexity the card represents is crucial.  They will be making assumptions that things may be easier than they are and as developers we're often best equipped to highlight anything they may not be aware of.
+
+Clarify what exactly completion means for the card.  You will need this to determine when you're done and when the code makes whatever improvement or fix that the client needs.  This information will inform how you test and verify completion.
+
+Ask if the client has any related metrics that can indicate a positive business change in the card.
+
+Before you move on to the next card you should have enough information to do the following:
+
+  -Make a checklist of things you need to do to complete the card
+  -Map out your testing strategy to verify any change to the system
+
+# Development
+
+* Update the project board to indicate you've started on the card.
+* Create a local feature branch in git based off master. Give your branch a short descriptive name based on the card on which you are working.
+* Review the checklist and update it based on what you need to do for card completion.
+* Map out your testing strategy for your card before you begin to change any code.  Refer to the [QA Mindset](https://github.com/HaughtCodeworks/TeamGuide) for guidance on the right level of testing.
+* Consider breaking down the steps for completion into smaller chunks allowing you to test and commit those changes more quickly, especially if they are isolated in different parts of the system.  Update your checklist as appropriate.
+* Work through each checklist item using test driven development to test, verify and complete that step.  Unless the step has no UI elements, manually test the change in your local environment.
```

Comment:
I'm wondering if we need to indicate that manual testing is done in the browser?  If so maybe something like this for the end:

Unless the step has no UI elements, manually test the change in the browser on your local environment.